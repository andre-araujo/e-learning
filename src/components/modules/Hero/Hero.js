import React from 'react';
import { string, node } from 'prop-types';

import CoverImage from '../../elements/CoverImage';

import {
  Title,
  Container,
  Content,
} from './styles';

function Hero({
  title,
  children,
  imageUrl,
  minHeight,
}) {
  return (
    <CoverImage imageUrl={imageUrl}>
      <Container minHeight={minHeight}>
        <Title>
          {title}
        </Title>
        <Content>
          {children}
        </Content>
      </Container>
    </CoverImage>
  );
}

Hero.defaultProps = {
  title: '',
  children: null,
  imageUrl: null,
  minHeight: null,
};

Hero.propTypes = {
  title: string,
  children: node,
  imageUrl: string,
  minHeight: string,
};

export default Hero;
