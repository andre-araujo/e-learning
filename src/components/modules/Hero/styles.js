import styled, { css } from 'styled-components';

import Wrapper from '../../elements/Wrapper';

export const Container = styled(Wrapper)`
  display: flex;
  align-items: center;
  justify-content: space-between;

  ${({ minHeight }) => !minHeight && css`
    min-height: 400px;
    height: calc(50vh);
  `}

  ${({ minHeight }) => minHeight && css`
    min-height: ${minHeight};
    height: auto;
  `}
`;

export const Title = styled.h1`
  color: white;
  font-weight: 500;
  font-size: 50px;
`;

export const Content = styled.div`
  font-size: 1rem;
`;
