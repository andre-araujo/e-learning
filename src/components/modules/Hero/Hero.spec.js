import React from 'react';
import { shallow } from 'enzyme';

import Hero from './Hero';

describe('Hero component', () => {
  it('should match snapshot', () => {
    expect(shallow(<Hero />)).toMatchSnapshot();
  });

  it('should match snapshot with title', () => {
    expect(shallow(<Hero title="Title" />)).toMatchSnapshot();
  });

  it('should match snapshot with children', () => {
    expect(shallow(
      <Hero>
        children
      </Hero>,
    )).toMatchSnapshot();
  });
});
