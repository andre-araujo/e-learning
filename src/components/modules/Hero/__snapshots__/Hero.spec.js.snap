// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component should match snapshot 1`] = `
<CoverImage
  imageUrl={null}
>
  <Styled(Wrapper)
    minHeight={null}
  />
</CoverImage>
`;

exports[`Hero component should match snapshot with children 1`] = `
<CoverImage
  imageUrl={null}
>
  <Styled(Wrapper)
    minHeight={null}
  >
    <styled.div>
      children
    </styled.div>
  </Styled(Wrapper)>
</CoverImage>
`;

exports[`Hero component should match snapshot with title 1`] = `
<CoverImage
  imageUrl={null}
>
  <Styled(Wrapper)
    minHeight={null}
  >
    <styled.div>
      Title
    </styled.div>
  </Styled(Wrapper)>
</CoverImage>
`;
